pilot/pkg/proxy/envoy/v2/ads.go

```diff
@@ -355,15 +368,21 @@ func (s *DiscoveryServer) removeCon(conID string, con *XdsConnection) {
 }
 
 func (s *DiscoveryServer) pushRoute(con *XdsConnection) error {
-	rc := [][]*xdsapi.RouteConfiguration{}
-	for _, rn := range con.Routes {
-		r, err := s.ConfigGenerator.BuildRoutes(s.env, *con.modelNode, rn)
-		if err != nil {
-			log.Warnf("ADS: config failure, closing grpc %v", err)
-			return err
-		}
+	rc := []*xdsapi.RouteConfiguration{}
+
+	var services []*model.Service
+	s.modelMutex.RLock()
+	services = s.services
+	s.modelMutex.RUnlock()
+
+	// TODO: once per config update
+	for _, routeName := range con.Routes {
+		// TODO: for ingress/gateway use the other method
+		r := s.ConfigGenerator.BuildSidecarOutboundHTTPRouteConfig(s.env, *con.modelNode, nil,
```

Comment:
this will break the gateway (and hence ingress) implementation immediately, since it will always fetch sidecar routes.

Reply:
ignore. its not activated